---
title: Derleyici Uyarısı (düzey 1) CS1058
ms.date: 07/20/2015
ms.prod: .net
ms.technology:
- devlang-csharp
ms.topic: error-reference
f1_keywords:
- CS1058
helpviewer_keywords:
- CS1058
ms.assetid: ed50590c-f130-47c3-976d-322a6c8f996d
caps.latest.revision: 11
author: BillWagner
ms.author: wiwagn
ms.openlocfilehash: 3d5d32b17c9803c4df82ff4e8cc3b0672a386b78
ms.sourcegitcommit: bd1ef61f4bb794b25383d3d72e71041a5ced172e
ms.translationtype: MT
ms.contentlocale: tr-TR
ms.lasthandoff: 10/18/2017
---
# <a name="compiler-warning-level-1-cs1058"></a><span data-ttu-id="fac33-102">Derleyici Uyarısı (düzey 1) CS1058</span><span class="sxs-lookup"><span data-stu-id="fac33-102">Compiler Warning (level 1) CS1058</span></span>
<span data-ttu-id="fac33-103">Önceki bir catch yan tümcesi zaten tüm özel durumları yakalar.</span><span class="sxs-lookup"><span data-stu-id="fac33-103">A previous catch clause already catches all exceptions.</span></span> <span data-ttu-id="fac33-104">Oluşturulan tüm özel durumları System.Runtime.CompilerServices.RuntimeWrappedException içinde sarılır</span><span class="sxs-lookup"><span data-stu-id="fac33-104">All exceptions thrown will be wrapped in a System.Runtime.CompilerServices.RuntimeWrappedException</span></span>  
  
 <span data-ttu-id="fac33-105">Bu öznitelik, CS1058 neden olan bir `catch()` blok sonra yazın hiçbir belirtilen özel durum oluştu bir `catch (System.Exception e)` bloğu.</span><span class="sxs-lookup"><span data-stu-id="fac33-105">This attribute causes CS1058 if a `catch()` block has no specified exception type after a `catch (System.Exception e)` block.</span></span> <span data-ttu-id="fac33-106">Uyarı öneren `catch()` bloğu değil tüm özel durumlar yakalayın.</span><span class="sxs-lookup"><span data-stu-id="fac33-106">The warning advises that the `catch()` block will not catch any exceptions.</span></span>  
  
 <span data-ttu-id="fac33-107">A `catch()` sonra engelleme bir `catch (System.Exception e)` catch bloğu CLS olmayan özel durumlar varsa `RuntimeCompatibilityAttribute` AssemblyInfo.cs dosyasında false olarak ayarlayın: `[assembly: RuntimeCompatibilityAttribute(WrapNonExceptionThrows = false)]`.</span><span class="sxs-lookup"><span data-stu-id="fac33-107">A `catch()` block after a `catch (System.Exception e)` block can catch non-CLS exceptions if the `RuntimeCompatibilityAttribute` is set to false in the AssemblyInfo.cs file: `[assembly: RuntimeCompatibilityAttribute(WrapNonExceptionThrows = false)]`.</span></span> <span data-ttu-id="fac33-108">Bu öznitelik açıkça false olarak ayarlanmamışsa, tüm atılmış CLS olmayan özel durumları özel durumlar olarak sarılır ve `catch (System.Exception e)` bloğu onları yakalar.</span><span class="sxs-lookup"><span data-stu-id="fac33-108">If this attribute is not set explicitly to false, all thrown non-CLS exceptions are wrapped as Exceptions and the `catch (System.Exception e)` block catches them.</span></span> <span data-ttu-id="fac33-109">Daha fazla bilgi için bkz: [nasıl yapılır: CLS olmayan özel Catch](../../../csharp/programming-guide/exceptions/how-to-catch-a-non-cls-exception.md).</span><span class="sxs-lookup"><span data-stu-id="fac33-109">For more information, see [How to: Catch a non-CLS Exception](../../../csharp/programming-guide/exceptions/how-to-catch-a-non-cls-exception.md).</span></span>  
  
## <a name="example"></a><span data-ttu-id="fac33-110">Örnek</span><span class="sxs-lookup"><span data-stu-id="fac33-110">Example</span></span>  
 <span data-ttu-id="fac33-111">Aşağıdaki örnek CS1058 oluşturur.</span><span class="sxs-lookup"><span data-stu-id="fac33-111">The following example generates CS1058.</span></span>  
  
```csharp  
// CS1058.cs  
// CS1058 expected  
using System.Runtime.CompilerServices;  
  
// the following attribute is set to true by default in the C# compiler  
// set to false in your source code to resolve CS1058  
[assembly: RuntimeCompatibilityAttribute(WrapNonExceptionThrows = true)]  
  
class TestClass   
{  
   static void Main()   
   {  
      try {}  
  
      catch (System.Exception e) {   
         System. Console.WriteLine("Caught exception {0}", e);  
      }  
  
      catch {}   // CS1058. This line will never be reached.  
   }  
}  
```
